clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -145,7 +155,16 @@ private SSLContext createSSLContext() throws GeneralSecurityException, IOExcepti
         KeyStore ts = truststore == null ? null : truststore.load();
         tmf.init(ts);
 
-        sslContext.init(keyManagers, tmf.getTrustManagers(), this.secureRandomImplementation);
+        TrustManager[] trustManagers = tmf.getTrustManagers();
+        TrustManager reloadableTrustManager = new ReloadableX509TrustManager(truststore, tmf);
```

Comment:
Is the reloadable trust manager always needed? The cases to consider are:
1. Broker using configured truststore (this is the case described in the JIRA)
2. Broker using default truststore: is reload useful here? should it be done for consistency?
3. Broker using SASL, or SSL without client-auth: reload should be disabled since the truststore is not used
3. Clients using configured or default truststore: JIRA doesn't talk about clients, should client truststores be reloaded as well (the code currently does).